source/speech/__init__.py

```diff
@@ -230,10 +230,64 @@ def speakSpelling(
 	speak(seq, priority=priority)
 
 
-# C901 'getSpellingSpeech' is too complex
-# Note: when working on getSpellingSpeech, look for opportunities to simplify
-# and move logic out into smaller helper functions.
-def getSpellingSpeech(  # noqa: C901
+def getSpellingSpeechAddCharMode(
+		seq: Generator[SequenceItemT, None, None],
+) -> Generator[SequenceItemT, None, None]:
+	"""Inserts CharacterMode commands in a speech sequence generator to ensure any single character
+	is spelt by the synthesizer.
+	@param seq: The speech sequence to be spelt.
+	"""
+	synth = getSynth()
+	synthConfig = config.conf["speech"][synth.name]
+	charMode = False
+	for item in seq:
+		if isinstance(item, str):
+			if len(item) == 1 and synthConfig["useSpellingFunctionality"]:
+				if not charMode:
+					yield CharacterModeCommand(True)
+					charMode = True
+			elif charMode:
+				yield CharacterModeCommand(False)
+				charMode = False
+		yield item
+
+
+def getSpellingCharWithCaseInfo(
+		speakCharAs: str,
+		uppercase: bool,
+		isPitchSupported: bool,
+) -> Generator[SequenceItemT, None, None]:
+	"""This function produces a speech sequence containing a character to be spelt as well as commands
+	to indicate this character case if applicable.
+	@param speakCharAs: The character as it will be spoken by the synthesizer.
+	@param uppercase: Indicates if the spelt character should be added uppercase notifications.
+	@param isPitchSupported: indicates if the current synthesizer supports pitch commands.
+	"""
+	synth = getSynth()
+	synthConfig = config.conf["speech"][synth.name]
+	
+	if uppercase and synthConfig["sayCapForCapitals"]:
+		# Translators: cap will be spoken before the given letter when it is capitalized.
+		capMsg = _("cap %s")
+		(capMsgBefore, capMsgAfter) = capMsg.split('%s')
+	else:
+		capMsgBefore = ''
+		capMsgAfter = ''
+	
+	if uppercase and isPitchSupported and synthConfig["capPitchChange"]:
+		yield PitchCommand(offset=synthConfig["capPitchChange"])
+	if uppercase and synthConfig["beepForCapitals"]:
+		yield BeepCommand(2000, 50)
+	if capMsgBefore:
+		yield capMsgBefore
+	yield speakCharAs
+	if capMsgAfter:
+		yield capMsgAfter
+	if uppercase and isPitchSupported and synthConfig["capPitchChange"]:
+		yield PitchCommand()
+
+
+def getSpellingSpeechWithoutCharMode(

```

Comment:
Please make this a private method for now:
```suggestion
def _getSpellingSpeechWithoutCharMode(

```